pennylane/templates/subroutines/double_excitation_unitary.py

```diff
@@ -0,0 +1,550 @@
+               R_z(\frac{\theta}{8}) \bigg], \\
+               && \bigg[H, R_x(-\frac{\pi}{2}), R_x(-\frac{\pi}{2}), R_x(-\frac{\pi}{2}),
+               R_z(\frac{\theta}{8}) \bigg], \bigg[H, H, H, R_x(-\frac{\pi}{2}),
+               R_z(\frac{\theta}{8}) \bigg], \\
+               && \bigg[R_x(-\frac{\pi}{2}), H, H, H, R_z(-\frac{\theta}{8}) \bigg],
+               \bigg[H, R_x(-\frac{\pi}{2}), H, H, R_z(-\frac{\theta}{8}) \bigg], \\
+               && \bigg[R_x(-\frac{\pi}{2}), R_x(-\frac{\pi}{2}), R_x(-\frac{\pi}{2}),
+               H, R_z(-\frac{\theta}{8}) \bigg], \bigg[R_x(-\frac{\pi}{2}), R_x(-\frac{\pi}{2}),
+               H, R_x(-\frac{\pi}{2}), R_z(-\frac{\theta}{8}) \bigg] \Bigg\}
+
+        #. For a given quadruple ``[s, r, q, p]`` with :math:`p>q>r>s`, seventy-two single-qubit
+           operations are applied. Notice also that consecutive CNOT gates act on qubits with
+           indices between ``s`` and ``r`` and ``q`` and ``p`` while a single CNOT acts on wires
+           ``r`` and ``q``. The operations performed across these qubits are shown in dashed lines
+           in the figure above.
+
+        An example of how to use this template is shown below:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import DoubleExcitationUnitary
+
+            dev = qml.device('default.qubit', wires=5)
+
+            @qml.qnode(dev)
+            def circuit(weight, pphh=None):
+                DoubleExcitationUnitary(weight, wires=pphh)
+                return qml.expval(qml.PauliZ(0))
+
+            weight = 1.34817
+            double_excitation = [0, 1, 3, 4]
+            print(circuit(weight, pphh=double_excitation))
+
+    """
+
+    ##############
+    # Input checks
+
+    check_no_variable(wires, msg="'wires' cannot be differentiable")
+
+    wires = check_wires(wires)
+
+    expected_shape = (4,)
+    check_shape(
+        wires,
+        expected_shape,
+        msg="'wires' must be of shape {}; got {}".format(expected_shape, get_shape(wires)),
+    )
+
+    expected_shape = ()
+    check_shape(
+        weight,
+        expected_shape,
+        msg="'weight' must be of shape {}; got {}".format(expected_shape, get_shape(weight)),
+    )
+
+    check_type(wires, [list], msg="'wires' must be a list; got {}".format(wires))
+    for w in wires:
+        check_type(w, [int], msg="'wires' must be a list of integers; got {}".format(wires))
+
+    if wires != sorted(wires):
+        raise ValueError(
+            "Elements of 'wires' must satisfy that wires_3 > wires_2 > wires_1 > wires_0.\n"
+            "Got wires[3]={}, wires[2]={}, wires[1]={}, wires[0]={}".format(*wires)
+        )
+
+    if (wires[1] <= wires[0]) or (wires[3] <= wires[2]):
+        raise ValueError(
+            "Elements of 'wires' must satisfy that wires_1 > wires_0 and wires_3 > wires_2.\n"
+            "Got wires[3]={}, wires[2]={}, wires[1]={}, wires[0]={}".format(*wires)
+        )
+
```

Comment:
Just a comment: All of these wire checks will be taken care of in the new wire management strategy.